Source/JavaScriptCore/assembler/ARM64Assembler.h

```diff
@@ -2908,9 +2910,9 @@ class ARM64Assembler {
     template<int dstsize, int srcsize>
     ALWAYS_INLINE void fcvt(FPRegisterID vd, FPRegisterID vn)
     {
-        ASSERT(dstsize == 16 || dstsize == 32 || dstsize == 64);
-        ASSERT(srcsize == 16 || srcsize == 32 || srcsize == 64);
-        ASSERT(dstsize != srcsize);
+        static_assert(dstsize == 16 || dstsize == 32 || dstsize == 64);
+        static_assert(srcsize == 16 || srcsize == 32 || srcsize == 64);
```

Comment:
Nit could be `CHECK_DATASIZE_OF_FLOATING_POINT(dstsize)`/`CHECK_DATASIZE_OF_FLOATING_POINT(srcSize)`

Reply:
Changed.